pkg/errors/base.go

```diff
@@ -127,10 +127,25 @@ func NewBaseError(kind ErrKind, errMsg string, err error, detail ErrDetails) Bas
 	}
 }
 
+// NewBaseErrorWrapper creates a new BaseError by wrapping an existing Error
+func NewBaseErrorWrapper(err Error) BaseError {
+	return NewBaseError(err.Kind(), "", err, err.Details())
+}
+
 // ToBaseError creates a new BaseError with Kind found from err
 func ToBaseError(err error) BaseError {
+	var (
+		be      BaseError
```

Comment:
```suggestion
		e      Error
```
it may be other Error type, so use the interface